### ISSUE TITLE
QGC bad at receiving waypoints in a different reference frame from what it sent

### ISSUE THREAD

**Issue by Susurrus** [mavlink/qgroundcontrol#1432]
Sometimes I send missions to my UAS as local coordinates. But whenever I have my UAS report its onboard missions, I only ever output them in the global frame. This is because QGC isn't good at displaying things not in the global frame. But if I load a waypoint set, upload it to the UAS, and then it reports back in global coordinates, these coordinates don't show up on the map. So I need to manually clear my mission list and re-request them from the UAS, even though it just sent them to me.

Basically QGC isn't very smart when it gets missions in a different coordinate frame than it sent them in. QGC should detect this and if any of them differ, throw out its onboard mission list and use the ones reported by the UAS.
